windows-agent/internal/daemon/testutils/testutils.go

```diff
@@ -0,0 +1,79 @@
+// Package testutils exports test helpers to be used in other packages that need to change internal behaviors of the daemon.
+package testutils
+
+import (
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+	"syscall"
+	"testing"
+
+	"github.com/canonical/ubuntu-pro-for-wsl/common/testdetection"
+)
+
+// MockWslSystemCmd mocks commands running inside the WSL system distro.
+// Add it to your package_test with:
+//
+//	func TestWithWslSystemMock(t *testing.T) { daemontests.MockWslSystemCmd(t) }
+//
+//nolint:thelper // This is a faux test used to mock commands running via `wsl --system`
+func MockWslSystemCmd(t *testing.T) {
+	testdetection.MustBeTesting()
+
+	const errorUsage = `
+wslinfo usage:
+	--networking-mode
+		Display current networking mode.
+
+	--msal-proxy-path
+		Display the path to the MSAL proxy application.
+
+	-n
+		Do not print a newline.
+	`
+
+	if os.Getenv("GO_WANT_HELPER_PROCESS") == "" {
+		t.Skip("Skipped because it is not a real test, but rather a mocked executable")
+	}
+
+	var argv []string
+	begin := slices.Index(os.Args, "--")
+	if begin != -1 {
+		argv = os.Args[begin+1:]
+	}
+
+	// Action
+	exit := func(args []string) int {
```

Comment:
I see what you did here, but as returning only do an exit, I would really just exit in the 2 cases we don’t return 0, and then don’t use a closure that you call immediatly.

That will save the finale `if` too by always printing in the "happy path"